ob1k-cache/src/main/java/com/outbrain/ob1k/cache/metrics/AsyncOperationMetrics.java

```diff
@@ -0,0 +1,40 @@
+package com.outbrain.ob1k.cache.metrics;
+
+import java.util.concurrent.TimeoutException;
+import com.outbrain.ob1k.concurrent.Try;
+import com.outbrain.swinfra.metrics.api.Counter;
+import com.outbrain.swinfra.metrics.api.MetricFactory;
+import com.outbrain.swinfra.metrics.api.Timer;
+
+/**
+ * Holds a set of metrics used for measuring cache operations.
+ *
+ * @author hunchback
+ */
+public class AsyncOperationMetrics {
+  private final Counter errors;
+  private final Counter timeouts;
+  private final Timer latency;
+
+  public AsyncOperationMetrics(final MetricFactory metricFactory, final String component) {
+    errors = metricFactory.createCounter(component, "errors");
+    timeouts = metricFactory.createCounter(component, "timeouts");
+    latency = metricFactory.createTimer(component, "latency");
+  }
+
+  public Timer.Context start() {
+    return latency.time();
```

Comment:
I have removed the timer.stop() from the finished(.., timer) method. As to the start() I did not want to enable direct access to  "latency" field ... thus I prefer getting the Timer.Context on behave of the caller. Any suggestions on an alternative?